torch/distributed/fsdp/flat_param.py

```diff
@@ -1573,31 +1561,30 @@ def _use_sharded_views(self) -> None:
             self._use_unsharded_views(as_params=True)
             return
         self._check_sharded(self.flat_param)
-        start, end = self.flat_param._shard_indices  # type: ignore[attr-defined]
+        start, end = self.flat_param._shard_indices
         offset = 0
-        assert self.flat_param._params is not None
+        # Construct once and reuse for all parameters not in the local shard
+        size_0_empty_tensor = torch.empty(
+            0,
+            dtype=self.flat_param.dtype,  # in case `flat_param` changed dtype
+            device=self.flat_param.device,
+            requires_grad=False,
+        )
         for i, (param, (param_name, module, _)) in enumerate(
             zip(self.flat_param._params, self.flat_param._param_infos)
         ):
             self._setattr_param(module, param_name, param)
             in_sharded_flat_param = (
-                i >= start
-                and i <= end
-                and self.flat_param._shard_param_offsets  # type: ignore[attr-defined]
+                i >= start and i <= end and self.flat_param._shard_param_offsets
             )
             if in_sharded_flat_param:
-                param_start, param_end = self.flat_param._shard_param_offsets[i - start]  # type: ignore[attr-defined]
+                param_start, param_end = self.flat_param._shard_param_offsets[i - start]
                 numel_in_shard = param_end - param_start + 1
                 param.data = self.flat_param[offset : offset + numel_in_shard]
                 offset += numel_in_shard
             else:
                 # Allow the original data to be freed via garbage collection
-                param.data = torch.empty(
-                    0,
-                    dtype=self.flat_param.dtype,  # in case `flat_param` changed dtype
-                    device=self.flat_param.device,
-                    requires_grad=False,
-                )
+                param.data = size_0_empty_tensor
```

Comment:
now each param's data will point to the same tensor's storage, I guess that's okay because this tensor doesn't really matter and is just used to allow the original data to be freed?